base_layer/core/src/consensus/consensus_constants.rs

```diff
@@ -63,6 +61,9 @@ pub struct ConsensusConstants {
     genesis_coinbase_value_offset: MicroTari,
     /// This is the maximum age a monero merge mined seed can be reused
     max_randomx_seed_height: u64,
+    /// This keeps track of the block split targets and which algo is accepted
+    /// Ideally this should count up to 100. If this does not you will reduce your target time.
+    algo_split: [u8; 3],
```

Comment:
Hmm, you're combining 2 functions in one field here. It's confusing and makes all the code below 5x more complicated than it needs to be.

I would actually just have an array of target block times here, and a function that returns the aggregate target block time, instead of explicitly targeting a given split.

Reply:
I am replaying to this below, as the single replay is covers both